django_auth0/auth_backend.py

```diff
@@ -5,27 +5,51 @@
 
 UserModel = get_user_model()
 
+# user profile keys that are always present as specified by
+# https://auth0.com/docs/user-profile/normalized#normalized-user-profile-schema
+AUTH0_USER_INFO_KEYS = [
+    'name',
+    'nickname',
+    'picture',
+    'user_id',
+]
+
 
 class Auth0Backend(object):
     def authenticate(self, **kwargs):
         """
         Auth0 return a dict which contains the following fields
-        :param email: user email
-        :param username: username
+        :param kwargs: user information provided by auth0
         :return: user
         """
-        email = kwargs.pop('email')
-        username = kwargs.pop('nickname')
-
-        if username and email:
-            try:
-                return UserModel.objects.get(email__iexact=email,
-                                             username__iexact=username)
-            except UserModel.DoesNotExist:
-                return UserModel.objects.create(email=email,
-                                                username=username)
-
-        raise ValueError(_('Username or email can\'t be blank'))
+        is_auth0 = True
+
+        # check that each auth0 key is present in kwargs
+        for key in AUTH0_USER_INFO_KEYS:
+            if key not in kwargs:
+                is_auth0 = False
+                break
+
+        # End the authentication attempt if this is not an auth0 payload
+        if is_auth0 is False:
+            return None
+
+        user_id = kwargs.get('user_id')
+
+        if user_id is None:
+            raise ValueError(_('user_id can\'t be blank!'))
```

Comment:
These two lines feel mismatching. If `user_id=''` then the `ValueError` won't be raised.